app/models/user.rb

```diff
@@ -95,6 +95,9 @@ def prompt_for_2sv?
 
   def event_logs
     EventLog.where(uid: uid).order(created_at: :desc)
+      .joins("LEFT JOIN ip_addresses ON ip_addresses.id = event_logs.ip_address_id")
+      .joins("LEFT JOIN user_agents ON user_agents.id = event_logs.user_agent_id")
```

Comment:
These joins should be doable using just ActiveRecord, you shouldn't need to write SQL for it